### DISCUSSION TITLE
MeshCentral and port 443

### DISCUSSION THREAD

**Discussion by cvpsmith** [Ylianst/MeshCentral#6067] [Q&A]
Not sure where to post this, but this looks like a good spot.  This is a general "Is this even possible?" type of question.  I have a Linux server set up with proper DNS, SSL, etc. which is running an Nginx server as proxy.  I also have Mattermost (OSS Slack-like server) running on port 8065 and Nginx re-routes 80 => 443 and does a proxy_pass of port 443 to 8065 for an SSL connection to Mattermost.

I'm trying to set up MeshCentral with Nginx so that the location "/mesh/" gets a proxy_pass from 443 to 4430, 80 to 800, etc. just like in the documentation.  Problem is, I can get to the MeshCentral login screen and login but, once I login, I get the message "Unable to connect web socket, click to reconnect." with the proper "MeshCentral" header with a black background at the bottom.  I see the leftmost bar ever so briefly when I click the login button, then it disappears and shows the black screen.

I've gone through all of the posts and set it up for websockets properly in my Nginx config:

proxy_pass http://127.0.0.1:4430/;
       proxy_http_version 1.1;
       # Allows websockets over HTTPS.
       proxy_set_header Upgrade $http_upgrade;
       proxy_set_header Connection "upgrade";
       proxy_set_header Host $host;
       proxy_read_timeout 86400;
       proxy_set_header Host $host;
       
       # Inform MeshCentral about the real host, port and protocol
       proxy_set_header X-Forwarded-Host $host:$server_port;
       proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
       proxy_set_header CF-Connecting-IP $proxy_add_x_forwarded_for;
       proxy_set_header X-Forwarded-Proto $scheme;

I've also run MeshCentral in debug mode and I see no errors.  But from what I've read, the MeshCentral clients need to access port 443.  So the question becomes, since the Mattermost server also uses port 443, is what I'm trying to do even possible?

Here's the relevant section of my  config.json:

   "settings": {
    "cert": "mesh.mydomain.com",
    "WANonly": true,
    "LANonly": false,
    "TlsOffload": "127.0.0.1",
    "_sessionKey": "myp@ssw0rd",
    "port": 4430,
    "aliasPort": 443,
    "redirPort": 800,
    "redirAliasPort":80,
    "cookieIpCheck": false,
    "AgentPong": 300
  },
  "domains": {
    "": {
      "certUrl": "https://127.0.0.1:443/",
      "title": "MeshCentral",
      "_title2": "MeshCentral Server",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },

Any help is GREATLY appreciated!  I'm basically looking for confirmation (or not!) that this is even possible.

**Comment by si458** [comment]
This might not work correctly because you are proxying https://mydomain.com/mesh/ to http://127.0.0.1:4430/mesh/ 

You need to get nginx to remove to the /mesh/ at the end for the inside proxy
OR
Create a new domain section in your config.json with "mesh" and then use that but that would mean all ur remote devices would talk to https://mydomain.com/mesh/

I think it would be better to use a subdomain instead and get nginx to reverse that to https://127.0.0.1:4430/

**Reply by cvpsmith** [reply]
> Create a new domain section in your config.json with "mesh" and then use that but that would mean all ur remote devices would talk to https://mydomain.com/mesh/

Hmm.  I tried setting up the domain like so:

  "domains": {
    "mesh": {
      "certUrl": "https://127.0.0.1:443/",
      "title": "NCCHE MeshCentral",

but I still get the same error.  Setting it up for https://mydomain.com/mesh/ wouldn't be an issue, but a subdomain isn't an option since I'm already down to the host level with DNS and the SSL cert.  So it's actually "https://myhost.mydomain.com/mesh/".  I don't have control over the DNS servers.  Do I need a new "mesh" domain in addition to the "" default domain, or do I just switch "" to "mesh"?  Also, thanks for the help!

**Reply by si458** [reply]
you would need both `"": { }` and `"mesh": { }` as they work as 2 different domains!
the example nginx config is here but this is using a base domain and not a sub folder
https://ylianst.github.io/MeshCentral/meshcentral/#nginx-reverse-proxy-setup

lets encrypt doews allows for `mesh1.mesh.mydomain.com`
so long as `mesh1.mesh.mydomain.com` or `*.mesh.mydomain.com` points to your servers IP

**Reply by cvpsmith** [reply]
I'll take a look and try these setting when I get a chance.  Thanks!

**Comment by hnsay** [comment]
Hello, have you found a solution for this?
